src/clj_sqs_extended/core.clj

```diff
@@ -72,43 +73,58 @@
   (.deleteQueue sqs-client url))
 
 (defn send-message
-  [sqs-client url message]
-  (.sendMessage sqs-client url message))
+  ([sqs-client url body]
+   (send-message sqs-client url body {}))
+  ([sqs-client url body
+    {:keys [format]
+     :or   {format :transit}
+     :as   opts}]
+   (let [payload (serdes/serialize body format)]
+     (.sendMessage sqs-client url payload))))
 
 (defn send-fifo-message
-  [sqs-client url message group-id]
-  (let [request (SendMessageRequest. url message)]
-    ; WATCHOUT: The group ID is mandatory when sending fifo messages.
-    (doto request (.setMessageGroupId group-id))
-    (.sendMessage sqs-client request)))
+  ([sqs-client url body group-id]
+   (send-fifo-message sqs-client url body group-id {}))
+  ([sqs-client url body group-id
+    {:keys [format]
+     :or   {format :transit}
+     :as   opts}]
+   (let [payload (serdes/serialize body format)
+         request (SendMessageRequest. url payload)]
+     ; WATCHOUT: The group ID is mandatory when sending fifo messages.
+     (doto request (.setMessageGroupId group-id))
+     (.sendMessage sqs-client request))))
 
 (defn delete-message
-  [sqs-client url message]
-  (let [request (DeleteMessageRequest. url (:receiptHandle message))]
+  [sqs-client url handle]
+  (let [request (DeleteMessageRequest. url (:receiptHandle handle))]
     (.deleteMessage sqs-client request)))
 
 (defn receive-message
   ([sqs-client url]
    (receive-message sqs-client url {}))
   ([sqs-client url
-    {:keys [wait-time
+    {:keys [format
+            wait-time
             visibility-timeout
             auto-delete]
-     :or   {wait-time 0}
+     :or   {format :transit
+            wait-time 0}
      :as   opts}]
    (letfn [(extract-relevant-keys [message]
              (-> (bean message)
                  (select-keys [:messageId :receiptHandle :body])))]
      (let [request (doto (ReceiveMessageRequest. url)
                      (.setWaitTimeSeconds (int wait-time))
                      (.setMaxNumberOfMessages (int 1)))
```

Comment:
1 is int already. you can drop the `int` @daisybytes. Also leave a design choice note as to why you're hardcoding this value to 1.

Reply:
Hmm.
(type 1)
=> java.lang.Long

Reply:
true. does it cause a problem to the client library passing a Long 1? Usually it's ok. Clojure/Java interop should be forgiving on types.

Reply:
Unfortunately the underlying library is very picky here and throws an exception without the additional ugly int.

Reply:
@daisybytes ahh ok